internal/commands/push/errors.go

```diff
@@ -8,3 +10,19 @@ func (err errProjectNotFound) Error() string {
 }
 
 func (err errProjectNotFound) DisableUsage() struct{} { return struct{}{} }
+
+type errNonExistantDirectory struct {
```

Comment:
so I think we can do all of this with just refactoring the existing `errProjectNotFound` type (which may want a bit of a rename)

we could define it as:

```go
type errProjectInvalid struct {
  path string // effectively searchPath from the 'inputs.go` file
  exists bool // indicating the path existed, but just wasn't a Realm app directory
}

func (err errProjectInvalid) Error() string {
  // logic based on `err.exists` for which error message to return
}
```